### PR TITLE
[SKU Recommendation] Adding telemetry for errors happening during data collection/ get  recommendation and telemetry

### PR DESCRIPTION
**Pull request by neetusingh-ms**
This change is important to understand what kind of errors are happening during data collection/ recommendation/ telemetry. This will help us improve our feature.

### CODE REVIEW

**Review by raymondtruong — APPROVED** [review]
Looks good to me - @brian-harris just wanted to make sure that these changes here are what you meant in this comment? https://github.com/microsoft/azuredatastudio/pull/18252#discussion_r805082297

**Review by brian-harris — APPROVED** [review]

File: extensions/sql-migration/src/models/stateMachine.ts [hunk 1]
```diff
@@ -394,7 +394,7 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 			}
 
 		} catch (error) {
-			console.log(error);
+			logError(TelemetryViews.SkuRecommendationWizard, 'GetSkuRecommendationFailed', error);
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/models/stateMachine.ts:397]: sendErrorEvent takes in a string for the errorCode as the 3rd parameter - not an error object 

https://github.com/microsoft/ads-extension-telemetry/blob/main/src/telemetryReporter.ts#L210

So I don't think this is going to do what you want.

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]

File: extensions/sql-migration/src/models/stateMachine.ts (hunk 1 above)
Comment (neetusingh-ms) [reply extensions/sql-migration/src/models/stateMachine.ts:397]: @Charles-Gagnon . for this release we are okay with just getting how many times error happened during a workflow. Till we figure out how to get an actual error (without PII).

Comment (Charles-Gagnon) [reply extensions/sql-migration/src/models/stateMachine.ts:397]: Sounds good, just remove the error parameter then and this should be good

Comment (neetusingh-ms) [reply extensions/sql-migration/src/models/stateMachine.ts:397]: export function logError(telemetryView: TelemetryViews, err: string, error: any): void {
	console.log(error);
	TelemetryReporter.sendErrorEvent(telemetryView, err);
}
error parameter is used for writing to console.

**Review by Charles-Gagnon — APPROVED** [review]

**Review by rchlkm — APPROVED** [review]

**Pull request merged**